src/server/routes/organisations/organisationSurveyController.js

```diff
@@ -91,7 +107,13 @@ const editOrganisationSurvey = async (req, res) => {
   const { user, body } = req
   const { id } = req.params
 
-  const updates = _.pick(body, ['name', 'startDate', 'endDate', 'teacherIds', 'studentNumbers'])
+  const updates = _.pick(body, ['name', 'startDate', 'endDate', 'teacherIds', 'studentNumbers', 'courseIds'])
+
+  const studentNumbersFromCourseIds = await getStudentNumbersFromCourseIds(updates.courseIds)
+
+  updates.studentNumbers = [
+    ...new Set([...updates.studentNumbers, ...studentNumbersFromCourseIds.filter(n => n !== null)]),
```

Comment:
Same filter trick as before, also if we are using the exact same thing here and there then maybe you should filter it in to a variable and use that to save some computation